test_pretend.py

```diff
@@ -159,3 +159,63 @@ def test_simple(self):
         f = call_recorder(lambda *args, **kwargs: 3)
         assert f() == 3
         assert f.calls == [call()]
+
+
+def _test_func(arg0, arg1, kwarg0=None, kwarg1=1):
+    pass
+
+
+class Klass:
+    def _test_method(self, arg0, arg1, kwarg0=None, kwarg1=1):
+        pass
+
+    @classmethod
+    def _test_class_method(cls, arg0, arg1, kwarg0=None, kwarg1=1):
+        pass
+
+    @staticmethod
+    def _test_static_method(arg0, arg1, kwarg0=None, kwarg1=1):
+        pass
+
+
+class TestSignature(object):
+    def test_bad_signature(self):
+        f = call_recorder(lambda *a, **kw: None, real_func=_test_func)
+        with pytest.raises(TypeError):
+            f(0)
+        assert f.calls == []
+
```

Comment:
If you add a test case for bad signature with `real_method`, I think you'll see it never raises :-)